### PR TITLE
Switch to get_post_modified_time for UTC

### LINKED ISSUE #88: lastmod time NOT in UTC
`<lastmod>2016-12-01T11:15:43Z</lastmod>`
The lastmod time is appended with 'Z' signifying it is the [ISO-8601 UTC timestamp](https://en.wikipedia.org/wiki/ISO_8601).
However, it is NOT the UTC time, but in actuality, the local time.

This can be easily fixed by modifying line [430 in msm-sitemap.php](https://github.com/Automattic/msm-sitemap/blob/master/msm-sitemap.php#L430) from 
`$url->addChild( 'lastmod', get_the_modified_date( 'Y-m-d' ) . 'T' . get_the_modified_date( 'H:i:s' ) . 'Z' );`
to
`$url->addChild( 'lastmod', get_post_modified_time( 'Y-m-dTH:i:s', true ) );`

### PR DESCRIPTION
**Pull request by mjangda**
Since we signify the datestamp as UTC via the z

props kimcheung

Fixes #88

### CODE REVIEW

File: msm-sitemap.php [hunk 1]
```diff
@@ -441,7 +441,7 @@ public static function generate_sitemap_for_date( $sitemap_date ) {
 
 			$url = $xml->addChild( 'url' );
 			$url->addChild( 'loc', get_permalink() );
-			$url->addChild( 'lastmod', get_the_modified_date( 'Y-m-d' ) . 'T' . get_the_modified_date( 'H:i:s' ) . 'Z' );
+			$url->addChild( 'lastmod', get_post_modified_time( 'Y-m-dTH:i:sZ', true ) );
```
Comment (nschoenholtz) [inline msm-sitemap.php]: @romancandlethoughts found a bug, and I think we've tracked it to this change -- would it be possible to use `c` format here? Or if not, the `T` needs to be escaped I think, and possibly the `Z` swapped for `P`? 

I'd be happy to make a PR if you'd like, but figured it'd make sense to start a discussion first. Thanks!

Comment (mjangda) [inline msm-sitemap.php]: I think `c` is probably good enough. A PR would be great.

Comment (mjangda) [inline msm-sitemap.php]: #101

Comment (romancandlethoughts) [inline msm-sitemap.php]: I was just about to update this, thanks @mjangda, you're the best!

**Comment by mjangda** [comment]
Here's a command that can be used to fix broken sitemaps created because of this issue: https://gist.github.com/mjangda/466bded7ad9de17169b1bccfc4ffd454

**Pull request merged**